### PR TITLE
feat: add EXCLUDE_AUTHORS env var for excluding users from conflict detection

### PR DESCRIPTION
**Pull request by zkoppert**
## Why

A user belonging to a `FILTER_TEAMS` team currently has no way to opt out of PR conflict detection without being removed from the team entirely. This creates friction when a team member needs to be excluded temporarily or permanently from scanning.

## What changed

I added a new `EXCLUDE_AUTHORS` environment variable that subtracts specific GitHub usernames from the combined filter set (after `FILTER_AUTHORS` and `FILTER_TEAMS` are resolved). This allows fine-grained exclusion without requiring team membership changes.

### Files modified

| File | Change |
|------|--------|
| `env.py` | Added `exclude_authors` field to `EnvVars` dataclass + parsing logic |
| `pr_conflict_detector.py` | Added step 2c: set subtraction after team resolution, with edge case handling |
| `pr_data.py` | Fixed truthiness bug: `if filter_authors:` → `if filter_authors is not None:` |
| `test_helpers.py` | Updated defaults + fixed mock truthiness check |
| `test_env.py` | Added 2 new tests, updated 3 existing assertions |
| `test_pr_conflict_detector_exclude.py` | New file with 5 integration tests for exclusion logic |
| `.env-example` | Added `EXCLUDE_AUTHORS=""` |
| `README.md` | Added env var documentation + usage example |

### Edge cases handled

| Scenario | Behavior |
|----------|----------|
| Exclude all filtered authors | Yields zero PRs scanned (not all) |
| EXCLUDE_AUTHORS without filters | Logs warning, scans all PRs (no filtering) |
| Non-matching usernames | Logs informational message |
| Empty sets after subtraction | Correctly handled via `is not None` checks |

### Tradeoffs

I chose to apply exclusion as a post-merge set subtraction (after FILTER_AUTHORS ∪ FILTER_TEAMS) rather than at the env parsing stage. This keeps the exclusion logic co-located with the filtering pipeline and makes it straightforward to log what was removed. The alternative of filtering during parsing would have been simpler but would lose visibility into the exclusion step.

## Testing

- 280 tests pass (99% coverage)
- 7 new tests: 2 in `test_env.py` (parsing), 5 in `test_pr_conflict_detector_exclude.py` (integration)
- Tests cover: team member exclusion, filter_authors exclusion, no-match warning, exclude-all-yields-zero, exclude-without-filters warning
- `make lint` passes clean (pylint 10/10, black, flake8, isort, mypy)

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

Closes #88

### CODE REVIEW

File: pr_conflict_detector.py [hunk 1]
```diff
@@ -85,7 +106,7 @@ def main():
             github_connection,
             owner,
             repo_name,
-            filter_authors=combined_filter_authors or None,
+            filter_authors=combined_filter_authors if filtering_requested else None,
         )
```
[the remark this answers is not in the document]
Comment (zkoppert) [reply pr_conflict_detector.py:113]: Good catch - this was a regression I introduced with the truthiness fix. When `FILTER_TEAMS` resolves to zero members and no `FILTER_AUTHORS` are set, `filtering_requested` was `True` (because the env var existed), causing an empty set to be passed and filtering out all PRs. That contradicted the existing warning message.

Fixed in dba994f by deriving `filtering_requested` from whether `combined_filter_authors` is non-empty after team resolution (but before exclusions). This preserves both behaviors:
- Empty team → `None` → scan all PRs (matches the warning)
- EXCLUDE_AUTHORS removes everyone → empty set → scan zero PRs (intended)

Added a regression test for the empty-team scenario.

**Review by jmeridth — APPROVED** [review]
